lib/internal/async_context_frame.js

```diff
@@ -23,6 +26,9 @@ class ActiveAsyncContextFrame extends SafeMap {
 
   static set(frame) {
     setContinuationPreservedEmbedderData(frame);
+    if (onSet.hasSubscribers) {
+      onSet.publish();
```

Comment:
You'll probably want to publish the frame object here. Also, you don't need the hasSubscribers check unless you're constructing a message object as publish is a no-op when there are no subscribers.

Reply:
We're exposing internals in plenty of other places too. I think it's generally fine here, and we might actually want to expose AsyncContextFrame publicly anyway. Regardless, all the channels are presently considered experimental and would typically also not be used except by those that at least mostly know what they are doing. I think exposing internals through an already somewhat internal event is a lot less critical than exposing internals on regular API surface. No one is going to discover this by accident and assume it's public, they'd have to find it through the docs.